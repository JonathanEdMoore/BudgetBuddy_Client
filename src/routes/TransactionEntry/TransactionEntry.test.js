import React from 'react'
import ReactDOM from 'react-dom'
import TransactionEntry from './TransactionEntry'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('TransactionEntry smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><TransactionEntry /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('TransactionEntry snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<TransactionEntry />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})