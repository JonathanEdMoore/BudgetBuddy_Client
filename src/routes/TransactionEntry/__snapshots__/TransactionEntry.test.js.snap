// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionEntry snapshot test renders the UI as expected 1`] = `
<div>
  <main
    role="main"
  >
    <header>
      <MainHeader />
      <h2>
        Transaction
      </h2>
    </header>
    <section>
      <BudgetBuddyForm
        className="signup-form"
        onSubmit={[Function]}
      >
        <div>
          <label
            htmlFor="transaction-type"
          >
            Deposit or Withdrawal?
          </label>
          <select
            name="type"
          >
            <option
              value="Deposit"
            >
              Deposit
            </option>
            <option
              value="Withdrawal"
            >
              Withdrawal
            </option>
          </select>
        </div>
        <div>
          <label
            htmlFor="title"
          >
            Title
          </label>
          <input
            id="title"
            name="title"
            placeholder="Name For transaction"
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="amount"
          >
            Amount
          </label>
          <input
            it="amount"
            name="amount"
            placeholder="10.00"
            tyep="number"
          />
        </div>
        <div>
          <label
            htmlFor="category"
          >
            Category
          </label>
          <select
            name="category"
          />
        </div>
        <button
          type="submit"
        >
          Submit
        </button>
      </BudgetBuddyForm>
    </section>
  </main>
</div>
`;
